cmd/swarm/swarm-smoke/pss.go

```diff
@@ -0,0 +1,237 @@
+		pssMessageCount = 1
+		log.Warn("message count should be a positive number. Defaulting to %d", pssMessageCount)
+	}
+
+	errc := make(chan error)
+	session := pssSetup()
+	defer func() {
+		for _, n := range session.nodes {
+			n.deregisterFunc()
+		}
+	}()
+
+	go func() {
+		var err error
+		for i := 0; i < pssMessageCount; i++ {
+			err = pssAsymDo(ctx, session, tuid)
+			if err != nil {
+				// Fail counter ++
+				metrics.GetOrRegisterCounter("pss.asym.failMsg", nil).Inc(1)
+				log.Error("error sending pss msg", "err", err)
+			} else {
+				// Success counter ++
+				metrics.GetOrRegisterCounter("pss.asym.successMsg", nil).Inc(1)
+			}
+		}
+		if metrics.GetOrRegisterCounter("pss.asym.failMsg", nil).Count() > 0 {
+			errc <- fmt.Errorf("some messages were not delivered")
+		} else {
+			errc <- nil
+		}
+	}()
+
+	select {
+	case err := <-errc:
+		if err != nil {
+			metrics.GetOrRegisterCounter("pss.asym.fail", nil).Inc(1)
+		}
+		return err
+	case <-time.After(time.Duration(timeout) * time.Second):
+		metrics.GetOrRegisterCounter("pss.asym.timeout", nil).Inc(1)
+		return fmt.Errorf("timeout after %v sec", timeout)
+	}
+
+}
+
+func pssSetup() *pssSession {
+	// random topic, one per session,  same for all msgs
+	topic := pss.BytesToTopic(testutil.RandomBytes(seed, 4))
+
+	log.Trace("pss random topic", "topic", topic.String())
+
+	session := &pssSession{
+		msgC:         make(chan pss.APIMsg),
+		requiredJobs: make(map[string]*pssJob),
+		allowedJobs:  make(map[string]*pssJob),
+		topic:        topic,
+	}
+
+	// set up the necessary info for each pss node
+	for i, host := range hosts {
+		httpHost := fmt.Sprintf("ws://%s:%d", host, 8546)
+		rpcClient, err := rpc.Dial(httpHost)
+		if err != nil {
+			log.Error("Error dialing host", "err", err)
+			continue
+		}
+
+		// get overlay address
+		var addr hexutil.Bytes
+		err = rpcClient.Call(&addr, "pss_baseAddr")
+		if err != nil {
+			log.Error("Error calling host for addr", "err", err)
+			continue
+		}
+
+		// get pss public key
+		var pubkey string
+		err = rpcClient.Call(&pubkey, "pss_getPublicKey")
+		if err != nil {
+			log.Error("Error calling host for pubkey", "err", err)
+			continue
+		}
+
+		// subscribe to the topic for the session
+		// this creates the incoming message handler automatically
+		// any message received with this topic comes on the channel
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+		defer cancel()
+		sub, err := rpcClient.Subscribe(ctx, "pss", session.msgC, "receive", session.topic, true, false)
+		if err != nil {
+			log.Error("Error calling host for subscribe", "err", err)
+			continue
+		}
+		session.nodes = append(session.nodes,
+			&pssNode{
+				hostIdx:        i,
+				addr:           []byte(addr),
+				client:         rpcClient,
+				pubkey:         pubkey,
+				deregisterFunc: sub.Unsubscribe,
+			},
+		)
+	}
+
+	return session
+}
+
+// sends a single PSS message between two random nodes using asymetric encryption
```

Comment:
comments in `godoc` start with the name of the method described.
`// pssAsymDo sends a single PSS message ...`